src/Features/ExternalAccess/OmniSharp/Internal/ExtractInterface/OmniSharpExtractInterfaceOptionsService.cs

```diff
@@ -5,53 +5,41 @@
 
-namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.Internal.ExtractInterface
-{
-    [Shared]
-    [ExportWorkspaceService(typeof(IExtractInterfaceOptionsService))]
-    internal class OmniSharpExtractInterfaceOptionsService : IExtractInterfaceOptionsService
-    {
-        private readonly IOmniSharpExtractInterfaceOptionsService _omniSharpExtractInterfaceOptionsService;
+namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.Internal.ExtractInterface;
 
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public OmniSharpExtractInterfaceOptionsService(IOmniSharpExtractInterfaceOptionsService omniSharpExtractInterfaceOptionsService)
-        {
-            _omniSharpExtractInterfaceOptionsService = omniSharpExtractInterfaceOptionsService;
-        }
+[ExportWorkspaceService(typeof(IExtractInterfaceOptionsService)), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class OmniSharpExtractInterfaceOptionsService(
+    IOmniSharpExtractInterfaceOptionsService omniSharpExtractInterfaceOptionsService) : IExtractInterfaceOptionsService
+{
+    private readonly IOmniSharpExtractInterfaceOptionsService _omniSharpExtractInterfaceOptionsService = omniSharpExtractInterfaceOptionsService;
 
-        public async Task<ExtractInterfaceOptionsResult> GetExtractInterfaceOptionsAsync(
-            ISyntaxFactsService syntaxFactsService,
-            INotificationService notificationService,
-            List<ISymbol> extractableMembers,
-            string defaultInterfaceName,
-            List<string> conflictingTypeNames,
-            string defaultNamespace,
-            string generatedNameTypeParameterSuffix,
-            string languageName,
-            CancellationToken cancellationToken)
-        {
-            var result = await _omniSharpExtractInterfaceOptionsService.GetExtractInterfaceOptionsAsync(extractableMembers, defaultInterfaceName).ConfigureAwait(false);
-            return new(
-                result.IsCancelled,
-                result.IncludedMembers,
-                result.InterfaceName,
-                result.FileName,
-                (ExtractInterfaceOptionsResult.ExtractLocation)result.Location);
-        }
+    public ExtractInterfaceOptionsResult GetExtractInterfaceOptions(
+        ISyntaxFactsService syntaxFactsService,
+        INotificationService notificationService,
+        List<ISymbol> extractableMembers,
+        string defaultInterfaceName,
+        List<string> conflictingTypeNames,
+        string defaultNamespace,
+        string generatedNameTypeParameterSuffix,
+        string languageName,
+        CancellationToken cancellationToken)
+    {
+        var result = _omniSharpExtractInterfaceOptionsService.GetExtractInterfaceOptionsAsync(extractableMembers, defaultInterfaceName).WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
same here.

note: this is no worse than before.  but now it's just isolated to the EA layer for omnisharp.

Reply:
Same as above.